### PR TITLE
Add start and end dates to transactions

### PR DESCRIPTION
**Pull request by AnnaGolosova**
## Link to ClickUp ticket

https://app.clickup.com/t/1k72d7k

## Describe this PR

* We do need an ability to filter transactions by startDate and endDate
* We do need an ability to get CreatedAt field from

### CODE REVIEW

**Review by HHjolany — APPROVED** [review]

File: HousingSearchApi/V1/Infrastructure/ElasticElasticSearchWrapper.cs [hunk 1]
```diff
@@ -17,16 +18,19 @@ public class ElasticSearchWrapper : IElasticSearchWrapper
         private readonly IQueryFactory _queryFactory;
         private readonly IPagingHelper _pagingHelper;
         private readonly ISortFactory _iSortFactory;
+        private readonly IFilterFactory _iFilterFactory;
```
Comment (amidomichael) [inline HousingSearchApi/V1/Infrastructure/ElasticElasticSearchWrapper.cs:21]: would rename to `_filterFactory` (and the same with the one above if possible) for consistency

File: HousingSearchApi/V1/Infrastructure/Filtering/FilterFactory.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace HousingSearchApi.V1.Infrastructure.Filtering
+{
+    public class FilterFactory : IFilterFactory
+    {
+        /// <summary>
+        /// Method to perform filtering from start to end dates
+        /// </summary>
+        public QueryContainer Filter<T, TRequest>(TRequest request, QueryContainerDescriptor<T> q) where T : class where TRequest : class
+        {
+            // ToDo: implement factory for choosing type of filter
```
Comment (amidomichael) [inline HousingSearchApi/V1/Infrastructure/Filtering/FilterFactory.cs:15]: is there a plan to do this?

Comment (AnnaGolosova) [reply HousingSearchApi/V1/Infrastructure/Filtering/FilterFactory.cs:15]: Yes, I will add this a little bit later, on the next week

File: HousingSearchApi/V1/Infrastructure/ElasticElasticSearchWrapper.cs (hunk 1 above)
Comment (AnnaGolosova) [reply HousingSearchApi/V1/Infrastructure/ElasticElasticSearchWrapper.cs]: Oh, sure. This is typo I guess. Already fixed

**Review by amidomichael — APPROVED** [review]

**Pull request merged**